setuptools/tests/test_build_meta.py

```diff
@@ -181,3 +182,13 @@ def test_build_sdist_version_change(build_backend):
 
     sdist_name = build_backend.build_sdist("out_sdist")
     assert os.path.isfile(os.path.join(os.path.abspath("out_sdist"), sdist_name))
+
+
+def test_build_meta_setup_py_exists(tmpdir_cwd):
+    # if build_sdist is called not from a script other than setup.py, ensure setup.py is included
```

Comment:
arguably this is not needed, if it's added to the egg I think the systems ensures it's packaged into the sdist.